libs/market-depth/src/lib/orderbook-controls.tsx

```diff
@@ -83,27 +72,78 @@ export const OrderbookControls = ({
                   isOpen ? VegaIconNames.CHEVRON_UP : VegaIconNames.CHEVRON_DOWN
                 }
               />
-              {formatResolution(resolution)}
+              {formatResolution(resolution, decimalPlaces)}
             </button>
           </TradingDropdownTrigger>
         }
       >
         <TradingDropdownContent align="start">
           {resolutions.map((r) => (
-            <TradingDropdownItem key={r} onClick={() => setResolution(r)}>
-              {formatResolution(r)}
+            <TradingDropdownItem
+              key={r}
+              onClick={() => setResolution(r)}
+              className="justify-end"
+            >
+              {formatResolution(r, decimalPlaces)}
             </TradingDropdownItem>
           ))}
         </TradingDropdownContent>
       </TradingDropdown>
       <button
         onClick={decreaseResolution}
         disabled={resolutions.indexOf(resolution) <= 0}
-        className="flex items-center px-2 cursor-pointer border-x border-default"
+        className="flex items-center px-2 cursor-pointer border-x border-default disabled:cursor-default"
         data-testid="minus-button"
       >
         <VegaIcon size={12} name={VegaIconNames.MINUS} />
       </button>
     </div>
   );
 };
+
+export const formatResolution = (r: number, decimalPlaces: number) => {
+  let num = addDecimalsFormatNumber(r, decimalPlaces);
+
+  // Remove trailing zeroes
+  num = num.replace(/(\.\d*?)0+$/, '$1');
```

Comment:
`.replace(/\.?0+/, '')` should remove trailing `0` and trailing `.`

Reply:
done